modules/password/pwn.go

```diff
@@ -0,0 +1,27 @@
+// Copyright 2020 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package password
+
+import (
+	"code.gitea.io/gitea/modules/setting"
+
+	"go.jolheiser.com/pwn"
+)
+
+// IsPwned checks whether a password has been pwned too many times
+// according to threshold
+func IsPwned(password string) (bool, error) {
+	if !setting.PasswordCheckPwn {
+		return false, nil
+	}
+
+	client := pwn.New()
+	count, err := client.CheckPassword(password, true)
+	if err != nil {
+		return true, err
```

Comment:
It's more usual to return false, err than true, err -> I know why you've done this but it's a gotcha that we're likely to mess up.

Reply:
I guess you could make a lot of noise in the header comment of the function to make it very clear that this function will return true with err.